### PR TITLE
controller/release: correct categories logic

### PR DESCRIPTION
**Pull request by mickeyn**
Code originally migrated from template. attempt to cut a corner was buggy.
This makes it a bit more accurate, but this code still need some love.

At least now the page will look like it does in v0.

### CODE REVIEW

**Comment by mickeyn** [comment]
this will solve #1767

File: lib/MetaCPAN/Web/Controller/Release.pm [hunk 1]
```diff
@@ -176,35 +176,42 @@ sub _files_to_categories {
...
-                    push @{ $ret->{provides} }, \%info;
-                }
+        my %info = (
+            status  => $f->{status},
+            path    => $f->{path},
+            release => $release->{name},
+            author  => $release->{author},
+        );
+
+        my @modules
```
Comment (ranguard) [inline lib/MetaCPAN/Web/Controller/Release.pm]: As this is the edge case, might be nice to add comment about the situation in which it happens, even add an e.g.?

Comment (mickeyn) [inline lib/MetaCPAN/Web/Controller/Release.pm]: as this is the first step of migrating code out of templates, i'd still wish to rework it as i don't find it good enough... just a 1-1 translation.

Comment (ranguard) [inline lib/MetaCPAN/Web/Controller/Release.pm]: For me ANY comments would be helpful - but JFDI is good as well. :)

**Pull request merged**